### PR TITLE
Api testing

### PR DESCRIPTION
**Pull request by pattyport19**
## Observations

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] The number of changes make it possible to review the code manually
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by jpdias** [comment]
@pattyport19 @AndreMalheiro98 Please remove `pipfile` dependency manager, since `requirements.txt` already accomplishes the required goal. (Just use `pip freeze > requirements.txt`, and, if required, `venv`).

**Comment by AndreMalheiro98** [comment]
@jpdias deleted pipenv. I'm not sure if dependencies are being correctly cached though

File: bin/handlers/admins.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import json
 
 def displayAdminPage():
+    #TODO - ONLY USERS WITH PROPERTY owner = True should see this page (as defined in the API documentation)
```
Comment (Esdeath-Shogun) [inline bin/handlers/admins.py:7]: A fix will be implemented this Tuesday (September, 13)

File: bin/handlers/licenses.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ def createLicense(productID, requestData):
         The return comes in a JSON format, made out of a 'code' field and a 'message' field. The function will always return an error as a 'code' if the productID is invalid or does not exist, if the request data is also invalid or if an error occurs while handling the Database. 
     """
     adminAcc = current_user
+    #TODO - replace json dumps when requirements are not matched. Response HTTP Code should be != 200
```
Comment (Esdeath-Shogun) [inline bin/handlers/licenses.py:38]: A fix will be implemented this Tuesday (September, 13)

File: bin/handlers/products.py [hunk 3]
```diff
@@ -15,6 +15,9 @@ def displayProduct(productID):
     if(not str(productID).isnumeric()):
         return Utils.render404("Product not found", "Sorry, but the product you have entered is invalid ...")
 
+    
+    # TODO - FIX BUG IF PRODUCT ID IS NOT IN DB
```
Comment (Esdeath-Shogun) [inline bin/handlers/products.py:19]: A fix will be implemented this Tuesday (September, 13)

File: bin/keys.py [hunk 4]
```diff
@@ -36,7 +36,8 @@ def generateSerialKey(length):
         serialKey += random.choice(characters)
     return serialKey
 
-# Data Format: licensekey-hardwareID 
+# Data Format: licensekey:hardwareID 
+#TODO - CATCH ERROR IF DECRYPTION FAILS. SERVER CRASHES
```
Comment (Esdeath-Shogun) [inline bin/keys.py:40]: The Server does not crash; only the worker does. This essentially means that the server would still be able to take in requests, but the specific request that experienced this fatal error would receive an 500 HTTP Code as a return.

A fix will be implemented this Tuesday (September, 13)

File: bin/handlers/licenses.py (hunk 2 above)
Comment (Esdeath-Shogun) [reply bin/handlers/licenses.py:38]: Fixed in the new merge request.

File: bin/keys.py (hunk 4 above)
Comment (Esdeath-Shogun) [reply bin/keys.py:40]: Fixed in the new merge request.

**Pull request merged**